flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -169,7 +169,7 @@ BufferAndBacklog pollBuffer(boolean isLocalChannel) {
 				BufferConsumer bufferConsumer = buffers.peek();
 
 				buffer = bufferConsumer.build();
-				if (!isLocalChannel && canBeCompressed(buffer)) {
+				if (!isLocalChannel && !bufferConsumer.isCopied() && canBeCompressed(buffer)) {
```

Comment:
nit: add a comment why "isCopied" or "isCopyable" `bufferConsumer` can not be compressed?

Reply:
I can replace the "isCopied" or "isCopyable" with "isBufferShared" which may be easier to understand.

Reply:
I think you are right. I think `isShareable` would be good enough to not require additional comments.